lucille-core/src/main/java/com/kmwllc/lucille/stage/FetchUri.java

```diff
@@ -0,0 +1,112 @@
+package com.kmwllc.lucille.stage;
+
+import com.kmwllc.lucille.core.ConfigUtils;
+import com.kmwllc.lucille.core.Document;
+import com.kmwllc.lucille.core.Stage;
+import com.kmwllc.lucille.core.StageException;
+import com.typesafe.config.Config;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.Iterator;
+import nl.altindag.ssl.util.UriUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.io.input.BoundedInputStream;
+import org.apache.http.HttpEntity;
+import org.apache.http.client.ClientProtocolException;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+
+public class FetchUri extends Stage {
+
+  private final String source;
+  private final String dest;
+  private final String statusSuffix;
+  private final String sizeSuffix;
+
+  private final String errorSuffix;
+  private final int maxDownloadSize;
+  private CloseableHttpClient client;
+
+  public FetchUri(Config config) {
```

Comment:
Should have some class-level javadoc saying what this stage does. Should mention any behaviors that might not be obvious e.g. the stage does nothing if the source field is absent on a document, or empty

Reply:
Added, also added javadoc for all config params and what they are for

Reply:
Great! But where does the doc mention what the stage does if the source field is not present on the doc, or is present but empty?